### PR TITLE
Rvv pre 1.0 2020 08 27

### PR DESCRIPTION
**Pull request by chihminchao**

### CODE REVIEW

**Comment by aswaterman** [comment]
Why remove quad-widening instructions?  Even if not in "V", they're still valid, right?

File: softfloat/fall_reciprocal.c [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+    return (UINT64_MAX >> (64 - len)) << pos;
+}
+
+//user needs to truncate output to required length
+static inline uint64_t rsqrte(uint64_t val, int e, int s, int p, bool sub) {
+  uint64_t exp = extract64(val, s, e);
+  uint64_t sig = extract64(val, 0, s);
+  uint64_t sign = extract64(val, s + e, 1);
+
+  const uint8_t table[] = {
```
Comment (aswaterman) [inline softfloat/fall_reciprocal.c]: You should make this `static const` so it doesn't get generated on the stack every time it's called.

**Comment by chihminchao** [comment]
The [spec update](https://github.com/riscv/riscv-v-spec/commit/72142aa47906ca4f92810d5a073aa046a394991e) has removed the opcode.  I think it should be invalid in 1.0 version

File: softfloat/fall_reciprocal.c (hunk 1 above)
Comment (chihminchao) [reply softfloat/fall_reciprocal.c]: good suggestion.  It will be fix in updated push.

File: softfloat/fall_reciprocal.c [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+  }
+
+  int idx = ((exp & 1) << (p-1)) | (sig >> (s-p+1));
+  uint64_t out_sig = (uint64_t)(table[idx]) << (s-p);
+  uint64_t out_exp = (3 * make_mask64(0, e - 1) + ~exp) / 2;
+
+  return (sign << (s+e)) | (out_exp << s) | out_sig;
+}
+
+float16_t f16_rsqrte(float16_t in, int precision)
```
Comment (aswaterman) [inline softfloat/fall_reciprocal.c]: I don't think precision should be an argument - it's implicit.

Comment (chihminchao) [reply softfloat/fall_reciprocal.c]: Do you means to use   "float16_t f16_rsqrte7(float16_t in)   and fix the precision in function ?

Comment (aswaterman) [reply softfloat/fall_reciprocal.c]: I think you have two options:
- change name to `f16_rsqrte7` and remove the parameter
- keep current name, and `assert(precision == 7);`

My preference is the first option.

Comment (aswaterman) [reply softfloat/fall_reciprocal.c]: > The [spec update](https://github.com/riscv/riscv-v-spec/commit/72142aa47906ca4f92810d5a073aa046a394991e) has removed the opcode. I think it should be invalid in 1.0 version

OK, I didn't notice that change.

File: softfloat/fall_reciprocal.c (hunk 1 above)
Comment (chihminchao) [reply softfloat/fall_reciprocal.c]: has been fixed

File: softfloat/fall_reciprocal.c (hunk 2 above)
Comment (chihminchao) [reply softfloat/fall_reciprocal.c]: I refine it as option 1

**Comment by chihminchao** [comment]
There is a issue to judge sNaN and qNaN case from fclassify and will be fixed in next push

**Comment by aswaterman** [comment]
@chihminchao ready to merge, or waiting for next push?

**Comment by chihminchao** [comment]
ready to merge.  Please use the latest push.

**Comment by chihminchao** [comment]
@aswaterman  
It is ok to merge.

**Pull request merged**